content/cif-connector/tests/src/libs/commerce/gui/components/common/cifproductfield/CifProductPickerTest.js

```diff
@@ -0,0 +1,224 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+'use strict';
+
+describe('CifProductPickerTest', () => {
+    var clickActivator = window.CIF.CifProductPicker.clickActivator;
+    var event = { preventDefault: function() {} };
+    var dollar = window.CIF.Granite.$;
+    var button;
+
+    before(() => {
+        var body = document.querySelector('body');
+        body.insertAdjacentHTML(
+            'afterbegin',
+            `<button class="js-editor-PageInfo-closePopover cq-commerce-cifproductpicker-activator cq-commerce-pdp-preview-activator coral3-Button coral3-Button--secondary" 
+                          title="View with Product" 
+                          type="button" 
+                          autocomplete="off" 
+                          is="coral-button" 
+                          trackingfeature="" 
+                          trackingelement="view with product" 
+                          tracking="ON" 
+                          size="M" 
+                          variant="secondary"><coral-button-label>View with Product</coral-button-label></button>`
+        );
+        button = document.querySelector('.cq-commerce-cifproductpicker-activator');
+
+        dollar.ajax = function(asd) {
```

Comment:
What's the purpose of this? `$.ajax()` returns a promise so you must properly mock that, i.e. something like 

```javascript
new Promise({resolve, reject} => return resolve(asd)}
```